Source/DafnyLanguageServer/Language/DafnyLangParser.cs

```diff
@@ -72,9 +73,23 @@ public sealed class DafnyLangParser : IDafnyParser, IDisposable {
         if (parseErrors != 0) {
           logger.LogDebug("encountered {ErrorCount} errors while parsing {DocumentUri}", parseErrors, document.Uri);
         }
+
         if (!TryParseIncludesOfModule(program.DefaultModule, program.BuiltIns, errorReporter, cancellationToken)) {
           logger.LogDebug("encountered error while parsing the includes of {DocumentUri}", document.Uri);
         }
+
+        return program;
+      } catch (Exception e) {
+        logger.LogDebug("encountered an exception while parsing {DocumentUri}", document.Uri);
+        logger.LogDebug(e.ToString());
```

Comment:
`LogDebug` has an [overload](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logdebug?view=dotnet-plat-ext-6.0#Microsoft_Extensions_Logging_LoggerExtensions_LogDebug_Microsoft_Extensions_Logging_ILogger_System_Exception_System_String_System_Object___) that accepts an exception as a parameter. It should print the stacktrace nicely:

```cs
logger.LogDebug(e, "encountered an exception while parsing {DocumentUri}", document.Uri);
```

Reply:
Nice one. I'll remember looking for overloads next time.